### PR TITLE
Adding Plugins Guide

### PR DESCRIPTION
**Pull request by JessicaSachs**
Outstanding:
- [x] Create `awesome-vue-test` contributed plugins (using the Testing section of awesome-vue)
- [x] Place some more examples of plugins and publish some of our own
- [x] Brainstorm on plugins we should write
- [x] Write initial docs on plugins

### CODE REVIEW

File: src/guide/plugins.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+Your plugin should be installed once. In Jest this should be in your Jest
+ config's `setupFiles` or `setupFilesAfterEnv` file.
+
+Some plugins automatically call `config.plugins.VueWrapper.install()` when
+ they're imported. This is common if they're extending multiple interfaces at
+  once. Follow the instructions of the plugin you're installing.
+
+Check out [awesome-vue-test]() for a collection of community-contributed
```
Comment (dobromir-hristov) [inline src/guide/plugins.md:46]: We can probably put these in https://vue-community.org/guide/ecosystem/testing.html

**Comment by JessicaSachs** [comment]
I don’t think that’s necessary. I don’t think Vue docs go into typescript
at all in the examples

File: src/guide/plugins.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Plugins add global-level functionality to Vue Test Utils' API. This is the
+ official way to extend Vue Test Utils' API with custom logic, methods, or
+   functionality.
+
+ If you're missing a bit of functionality, consider writing a plugin to
+  extend Vue Test Utils' API.
+
+ Some use cases for plugins:
+ 1. Aliasing existing public methods
+ 1. Attaching matchers to the Wrapper instance
```
Comment (afontcu) [inline src/guide/plugins.md:12]:
```suggestion
 2. Attaching matchers to the Wrapper instance
```

File: src/guide/plugins.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ 1. Attaching functionality to the Wrapper or WrapperArray
```
Comment (afontcu) [inline src/guide/plugins.md:13]:
```suggestion
 3. Attaching functionality to the Wrapper or WrapperArray
```

File: src/guide/plugins.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+// locally defined plugin, see "Writing a Plugin"
+import MyPlugin from './myPlugin'
+
+// Install a plugin onto VueWrapper
+config.plugins.VueWrapper.install(MyPlugin)
+```
+
+You can optionally pass in some options:
+```js
+config.plugins.VueWrapper.install(MyPlugin, { someOption: true })
```
Comment (afontcu) [inline src/guide/plugins.md:36]: Should we showcase why options may come in handy? (we can do that in another PR, no worries). Have you thought about any use case?

File: src/guide/plugins.md (hunk 2 above)
Comment (JessicaSachs) [reply src/guide/plugins.md:12]: Markdown renders 1's sequentially so when you re-order lists you don't have to keep updating your numbers.
Screenshot 2020-04-26 04 27 08

File: src/guide/plugins.md (hunk 4 above)
Comment (JessicaSachs) [reply src/guide/plugins.md:36]: We were thinking about requiring the name for better error handling, otherwise it'll be any use-case where the user provides configuration options.

@lmiller1990 has written a few plugins already and I hope to feature them as examples at the beginning of the intro

Comment (lmiller1990) [reply src/guide/plugins.md:36]: I like the idea of having a name if it helps with error messages. But I don't mind.

I will post my plugins soon! Let me clean them up a tiny bit.

File: src/guide/plugins.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  once. Follow the instructions of the plugin you're installing.
+
+Check out the [Vue Community Guide](https://vue-community.org/guide/ecosystem
+/testing.html) or [awesome-vue](https://github.com/vuejs/awesome-vue#test
+) for a collection of community-contributed plugins and libraries.
+
+## Writing a Plugin
+
+A Vue Test Utils plugin is simply a function that receives the mounted
+ `VueWrapper` or `DomWrapper` instance and can modify it.
```
Comment (lmiller1990) [inline src/guide/plugins.md:53]: In other locations we called `DOMWrapper`. I don't mind which we go with, but we should pick one

File: src/guide/plugins.md (hunk 4 above)
Comment (lmiller1990) [reply src/guide/plugins.md:36]: Ok, here is the plugin I have been using in my personal projects (needs VTU alpha-4 to be released to have correct typings)

```ts
const DataTestIdPlugin = (wrapper: VueWrapper<ComponentPublicInstance>) => {
  function findByTestId(selector: string): DOMWrapper<Element> | DOMWrapperError {
    const dataSelector = `[data-testid='${selector}']`
    const element = wrapper.element.querySelector(dataSelector)
    if (element) {
      return new DOMWrapper(element)
    }

    return new DOMWrapperError({ selector: dataSelector })
  }

  return {
    findByTestId
  }
}

config.plugins.VueWrapper.install(DataTestIdPlugin)
```

And I got typings like this:

```ts
declare module '@vue/test-utils' {
  interface VueWrapper<T extends ComponentPublicInstance> {
    findByTestId: (selector: string) => DOMWrapper<Element> | DOMWrapperError
  }
}
```

Comment (lmiller1990) [reply src/guide/plugins.md:36]: Although I think we use JS for the docs; TS add a little too much complexity. TS people know what to do.

Comment (JessicaSachs) [reply src/guide/plugins.md:36]: I think we should generally use JS for the docs. I'm going to keep this one example in though.
